File: clients/python/test/local_smoke.py
"""Runtime smoke for in-process (managed subprocess) mode. `local()` spawns and
owns a `pawrly console` child — no external orchestration needed.

    PYTHONPATH=src PAWRLY_BIN=../../target/debug/pawrly python3 test/local_smoke.py
"""

import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))

import pawrly  # noqa: E402

binary = os.environ.get("PAWRLY_BIN", "pawrly")

with pawrly.PawrlyClient.local(binary=binary) as c:
    assert c.transport == "local", c.transport
    res = c.query("SELECT 1 AS n").collect()
    assert res.row_count == 1 and res.rows[0]["n"] == 1, res
    out = c.materialize("local_mat", pawrly.MaterializeSpec(sql="SELECT 1 AS n"))
    assert out.name["table"] == "local_mat", out
    back = c.query("SELECT * FROM materialized.local_mat").collect()
    assert back.row_count == 1, back

print("LOCAL SMOKE OK")

File: clients/typescript/test/local_smoke.mjs
// Runtime smoke for in-process (managed subprocess) mode. `local()` spawns and
// owns a `pawrly console` child — no external orchestration needed.
//   PAWRLY_BIN=../../target/debug/pawrly node test/local_smoke.mjs
import { PawrlyClient } from "../dist/index.js";

const binary = process.env.PAWRLY_BIN ?? "pawrly";
const client = await PawrlyClient.local({ binary });
try {
  if (client.transport !== "local") throw new Error(`transport ${client.transport}`);
  const res = await (await client.query("SELECT 1 AS n")).collect();
  if (res.rowCount !== 1 || res.rows[0]?.n !== 1) {
    throw new Error(`query ${JSON.stringify(res)}`);
  }
  const out = await client.materialize("local_mat_ts", {
    kind: "query",
    sql: "SELECT 1 AS n",
  });
  if (out.name.table !== "local_mat_ts") {
    throw new Error(`materialize ${JSON.stringify(out.name)}`);
  }
  const back = await (
    await client.query("SELECT * FROM materialized.local_mat_ts")
  ).collect();
  if (back.rowCount !== 1) throw new Error(`readback ${back.rowCount}`);
  console.log("LOCAL SMOKE OK");
} finally {
  client.close();
}
